packages/conform-validitystate/index.ts

```diff
@@ -697,7 +698,7 @@ export function parse<
 		});
 	}
 
-	const value = Object.fromEntries(valueMap) as any;
+	const value = Object.fromEntries(valueMap) as InferType<Constraints>;
```

Comment:
this is also a little bit of a lie, I currently just forced it to whatever the used variable is expecting there's also a case where it expects both a Partial<T> and T but didn't make sense of the full context, any suggestions here would be helpful.